pirania-app/files/www/portal/js/voucher.js

```diff
@@ -35,98 +42,104 @@ function authVoucher () {
     id: 99,
     jsonrpc: '2.0',
     method: 'call',
-    params:[
+    params: [
       '00000000000000000000000000000000',
       'pirania',
       'auth_voucher',
       {
         voucher,
-        mac,
-      },
+        mac
+      }
     ]
   }
   fetch(url, {
     method: 'POST',
     body: JSON.stringify(authVoucherForm),
     headers: {
       'Access-Control-Allow-Origin': 'http://thisnode.info'
-    },
+    }
   })
-  .then(parseJSON)
-  .then((res) => {
-    hide(loader)
-    // debugger
-    if (res && res.result[1] && res.result[1].success) {
-      result.innerHTML = int[lang].success
-      show(result)
-      hide(errorElem)
-      init()
-      const urlTo = window.location.href.split('=')[1]
-      if (urlTo) {
-        window.location.href = `http://${urlTo}`
+    .then(parseJSON)
+    .then(res => {
+      hide(loader)
+      // debugger
+      if (res && res.result[1] && res.result[1].success) {
+        result.innerHTML = int[lang].success
+        show(result)
+        hide(errorElem)
+        init()
+        const urlTo = window.location.href.split('=')[1]
+        if (urlTo) {
+          window.location.href = `http://${urlTo}`
+        }
+      } else if (res && res.result[1] && !res.result[1].success) {
+        errorElem.innerHTML = int[lang].invalid
+        show(errorElem)
+      } else if (res.error) {
+        console.log(res.error)
+        errorElem.innerHTML = res.error
+        show(errorElem)
+
+        ubusError = true
       }
-    } else if (res && res.result[1] && !res.result[1].success) {
-      errorElem.innerHTML = int[lang].invalid
-      show(errorElem)
-    } else if (res.error) {
-      console.log(res.error)
-      errorElem.innerHTML = res.error
+      voucherElem.value = ''
+    })
+    .catch(err => {
+      console.log('Erro no Ubus', err)
+      errorElem.innerHTML = err
       show(errorElem)
-
       ubusError = true
-    }
-    voucherElem.value = ''
-  })
-  .catch((err) => {
-    console.log('Erro no Ubus', err)
-    errorElem.innerHTML = err
-    show(errorElem)
-    ubusError = true
-  })
-
+    })
 }
 
-function getIp() {
-  window.RTCPeerConnection = window.RTCPeerConnection || window.mozRTCPeerConnection || window.webkitRTCPeerConnection;//compatibility for Firefox and chrome
-  var pc = new RTCPeerConnection({iceServers:[]}), noop = function(){};      
-  pc.createDataChannel('');//create a bogus data channel
-  pc.createOffer(pc.setLocalDescription.bind(pc), noop);// create offer and set local description
-  pc.onicecandidate = function(ice) {
+function getIp () {
```

Comment:
I think it is convenient not to use WebRTC to obtain the IP. The reason is because the device may have other network connections than those necessarily connected to the router (vpn, mobile internet connections, etc). The address we need is the one assigned by the router's DHCP. It is best to use the http request.

Reply:
Good point. But this is what I'm getting from the `client_ip` script:

```
mac:
valid: false
ip: "2a00:1508:a93:9100:bb05:df94:cc1c:a411"
```

Needs fixing to be used.

Reply:
Okay, that's the bug to be solved then. So I saw quickly from the client_ip script I detected two problems:
* arp only exposes the ipv4, that we want to look for the mac with an ipv6 is not going to work that way
* client_ip scripts does not have execution permission. I had to assign it by hand in my router

Reply:
Yea, noticed the second problem, sending a 755 permission with the PR.

Let's work on this on another PR then.